### PR TITLE
LibGfx/JPEG2000: Use streams instead of manual offsets + minor cleanups

### PR DESCRIPTION
**Pull request by nico**

### CODE REVIEW

**Comment by nico** [comment]
Up to you if you prefer the 3rd commit. Either merge as is or tell me to drop it :)

File: Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp [hunk 1]
```diff
@@ -90,8 +90,8 @@ static ErrorOr<StartOfTilePart> read_start_of_tile_part(ReadonlyBytes data)
         return Error::from_string_literal("JPEG2000ImageDecoderPlugin: Not enough data for SOT marker segment");
 
     StartOfTilePart sot;
-    sot.tile_index = *reinterpret_cast<AK::BigEndian<u16> const*>(data.data());
-    sot.tile_part_length = *reinterpret_cast<AK::BigEndian<u32> const*>(data.data() + 2);
+    sot.tile_index = *reinterpret_cast<BigEndian<u32> const*>(data.data());
```
Comment (trflynn89) [inline Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp]: Is that u32 change intentional? `tile_index` is still u16

**Review by trflynn89 — COMMENTED** [review]
Cool!

File: Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp]: Not intentional. A bad day for our zero-bugs policy 😪

Comment (trflynn89) [reply Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp]: I wonder if we should invent some stream method thing that makes this more bugproof - just kinda pseudo-coding here:
```c++
// In stream:
template <typename T>
ErrorOr<void> read_value_into(T& value)
{
    value = TRY(read_value<T>());
}

template <typename EndianType, typename T> // EndianType might need to be "template template"
ErrorOr<void> read_value_into(T& value)
{
    value = TRY(read_value<EndianType<T>>());
}

// In user code:
TRY(stream.read_value_into<BigEndian>(sot.tile_index));
TRY(stream.read_value_into(sot.tile_part_index)); // is a u8
```

**Pull request merged**